src/Uno.Toolkit.UI/Extensions/DependencyObjectExtensions.cs

```diff
@@ -231,7 +231,10 @@ internal static void SetParent(this DependencyObject dependencyObject, object? p
 				{
 					if (dpInfo is { } && GetValue(dpInfo) is DependencyProperty dp)
 					{
-						var holderProperty = dpInfo.DeclaringType?.GetProperty(propertyName, Public | NonPublic | Instance);
+						// DeclaredOnly: Specifies flags that control binding and the way in which the search for members and types is conducted by reflection.
+						// because 'dpInfo.DeclaringType' is the guaranteed type, and we don't want a overridden property from a random base to throw AmbiguousMatchException
```

Comment:
```suggestion
						// because 'dpInfo.DeclaringType' is the guaranteed type, and we don't want an overridden property from a random base to throw AmbiguousMatchException
```